File: data/src/main/java/org/cryptomator/data/cloud/crypto/DirIdCache.java
package org.cryptomator.data.cloud.crypto;

import org.cryptomator.domain.CloudFolder;

interface DirIdCache {

	DirIdInfo get(CryptoFolder folder);

	DirIdInfo put(CryptoFolder folder, DirIdInfo dirIdInfo);

	void evict(CryptoFolder folder);

	void evictSubFoldersOf(CryptoFolder cryptoFolder);

	class DirIdInfo {

		private final String id;
		private final CloudFolder cloudFolder;

		DirIdInfo(String id, CloudFolder cloudFolder) {
			this.id = id;
			this.cloudFolder = cloudFolder;
		}

		public String getId() {
			return id;
		}

		public CloudFolder getCloudFolder() {
			return cloudFolder;
		}

		DirIdInfo withCloudFolder(CloudFolder cloudFolder) {
			return new DirIdInfo(id, cloudFolder);
		}
	}

}

File: data/src/main/java/org/cryptomator/data/cloud/crypto/DirIdCache.kt
package org.cryptomator.data.cloud.crypto

import org.cryptomator.domain.CloudFolder

interface DirIdCache {

	operator fun get(folder: CryptoFolder): DirIdInfo?

	fun put(folder: CryptoFolder, dirIdInfo: DirIdInfo): DirIdInfo

	fun evict(folder: CryptoFolder)

	fun evictSubFoldersOf(cryptoFolder: CryptoFolder)

	class DirIdInfo internal constructor(val id: String, val cloudFolder: CloudFolder) {

		fun withCloudFolder(cloudFolder: CloudFolder): DirIdInfo {
			return DirIdInfo(id, cloudFolder)
		}
	}
}
